devtools/cli/src/main/java/io/quarkus/cli/RegistryRemoveCommand.java

```diff
@@ -31,26 +32,29 @@ public Integer call() throws Exception {
         }
 
         final RegistriesConfig.Mutable config;
-        if (existingConfig) {
-            registryClient.refreshRegistryCache(output);
-            config = registryClient.resolveConfig().mutable();
-            if (config.getSource().getFilePath() == null) {
-                output.error("Can only modify file-based configuration. Config source is " + config.getSource().describe());
-                return CommandLine.ExitCode.SOFTWARE;
-            }
+        if (configYaml != null && !existingConfig) {
+            // we're creating a new configuration for a new file
+            config = RegistriesConfig.builder();
```

Comment:
Does it make sense for remove? Shouldn't the error removed below be moved here instead?

Reply:
In the interest of avoiding a lot of unnecessary null checks, I made this more similar to what happens with add. So if that condition is true, you'd attempt to remove a registry from the default configuration, and would write the result to the specified location (which would cause the file to be created). While perhaps not particularly useful, it isn't an error.